### ISSUE TITLE
`verusfmt` doesn't handle borrows of identifiers that start with the characters `mut`

### ISSUE THREAD

**Issue by pratapsingh1729** [jaybosamiya/verusfmt#6]
Input:
```
verus! {
pub exec fn foo(mut_state: &mut Blah) {
    let a = { b(&mut_state.c) };
}
} // verus!
```
Error:
```
Error:  --> 7:27
  |
7 |     let a = { b(&mut_state.c) };
  |                           ^---
  |
  = expected COMMENT, bang_str, colons_str, generic_arg_list, or record_expr_field_list
```
The error goes away and the code is correctly formatted if I do any of the following:
- remove the borrow: `let a = { b(mut_state.c) };`
- remove the characters `mut_`: `let a = { b(&state.c) };`
- prepend something before `mut`: `let a = { b(&xmut_state.c) };`

I would guess this is something to do with the parsing of `&mut`? `rustfmt` is able to handle the analogous Rust code:
```
pub fn foo(mut_state: &mut Blah) {
    let a = { b(&mut_state) };
}
```

**Comment by jaybosamiya** [comment]
This seems to be the same sort of problem as https://github.com/jaybosamiya/verusfmt/commit/4a9d96cd8131edba716a1912d5dddd0e78028d1d

We could fix it the same way, but I'm trying to think of a way to fix this once and for all, rather than just fixing each individual instance as we hit it. If this is blocking for you, do let me know and I'll push the relevant fix, but if it is not blocking, I'd rather we have a proper fix to this style of issue

**Comment by jaybosamiya** [comment]
Ok, figured out a way to fix this once and for all (see #8); once that is merged, that will fix this issue.
